tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/util/ServerUtil.java

```diff
@@ -487,23 +488,58 @@ public static PortalBundle getPortalBundle(IProject project)  throws CoreExcepti
 
         IStatus status = sdk.validate();
 
-        if ( !status.isOK())
+        if ( !status.isOK() )
         {
             return null;
         }
 
+        final String cachedAppServerPath = LiferayServerCore.getPreference(sdk.getName() + ".server.path");
```

Comment:
Where are these caches cleared out?  When the user changes the settings in build.properties, are these preferences clear?  Or are they simply set again to the correct values?